integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/noarkexchange/receive/InternalQueue.java

```diff
@@ -109,11 +113,31 @@ public void externalListener(byte[] message, Session session) {
                 }
             }
         } catch (Exception e) {
-            Audit.error("Failed to send message... queue will retry", EDUCoreMarker.markerFrom(request));
+            Audit.warn("Failed to send message... queue will retry", EDUCoreMarker.markerFrom(request));
             throw e;
         }
     }
 
+    /**
+     * Log failed messages as errors
+     */
+    @JmsListener(destination = DLQ)
+    public void dlqListener(byte[] message, Session session) {
+
+        try {
+            EDUCore request = eduCoreConverter.unmarshallFrom(message);
+            Audit.error("Failed to send message. Moved to DLQ", EDUCoreMarker.markerFrom(request));
+        } catch (Exception e) {
+        }
+
+        try {
+            EduDocument eduDocument = documentConverter.unmarshallFrom(message);
+            Audit.error("Failed to forward message. Moved to DLQ.", eduDocument.createLogstashMarkers());
+        } catch (Exception e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1166)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either remove or fill this block of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS00108)